src/lib.rs

```diff
@@ -1,53 +1,117 @@
+//! Inquirer – Fancy user interaction on the command line
+
+// Termion depends on nighty (for Keys), we need this for the unstable `std::io::CharsError` 
+#![feature(io)]
+
+#![deny(missing_docs,
+    missing_debug_implementations, missing_copy_implementations,
+    trivial_casts, trivial_numeric_casts,
+    unsafe_code,
+    unused_import_braces, unused_qualifications)]
+
 extern crate termion;
+#[macro_use] extern crate quick_error;
 
-use termion::{TermRead, TermWrite, IntoRawMode, color, Style, Key};
 use std::io::{Write, stdout, stdin};
+use std::fmt::Display;
+use termion::{TermRead, TermWrite, IntoRawMode, color, Style, Key};
+
+/// An option the user can choose
+///
+/// (Since the name "Option" is reserved for the well-known type representing
+/// nullability, we are calling this one "Choice".)
+pub trait Choice {
+    /// User visible text
+    type Text: Display;
+    /// Internal value representing this choice
+    type Value;
+
+    /// Get a reference to the text
+    fn text(&self) -> &Self::Text;
+
+    /// Get a reference to the value of this choice
+    fn value(&self) -> &Self::Value;
+}
+
+impl<'a> Choice for &'a str {
+    type Text = &'a str;
+    type Value = &'a str;
+
+    fn text(&self) -> &Self::Text {
+        self
+    }
+
+    fn value(&self) -> &Self::Value {
```

Comment:
I'm imagining a use-case where the user has to choose between a couple of different options, but the application programmer is too lazy to make separate enums, or it doesn't make sense.

I'm imagining a general use case like the following:
```rust
enum Color { Red, Blue, Green };

let choice = inquirer::list("Choose a color",
    &[("Red color", Color::Red),
    ("Blue color", Color::Blue),
    ("Green color", Color::Green)]);
```

What I had in mind with `simple_list` was, that sometimes the programmer might be too lazy to actually create separate enums for each menu, and would instead opt for the simpler:

```rust
let choice = inquirer::simple_list("Choose a color",
    &["Red color",
    "Blue color",
    "Green color"]);

match x {
    1 => ...
    ...
}
```

And in that case, I think it makes more sense to return numbers.

Perhaps I'm overthinking it? Surely, your enumeration example would work fine for my lazy programmer.